switmt_to_iso20022/data_mapper.bal

```diff
@@ -0,0 +1,3795 @@
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/data.xmldata;
+import ballerina/time;
+import ballerina/uuid;
+
+import ballerinax/financial.iso20022.cash_management as camtIsoRecord;
+import ballerinax/financial.iso20022.payment_initiation as painIsoRecord;
+import ballerinax/financial.iso20022.payments_clearing_and_settlement as pacsIsoRecord;
+import ballerinax/financial.swift.mt as swiftmt;
+
+# Converts a SWIFT message in string format to its corresponding ISO 20022 XML format.
+# 
+# The function uses a map of transformation functions corresponding to different SWIFT MT message types 
+# and applies the appropriate transformation based on the parsed message type.
+#
+# + finMessage - The SWIFT message string that needs to be transformed to ISO 20022 XML.
+# + return - Returns the transformed ISO 20022 XML or an error if the transformation fails.
+public isolated function toIso20022Xml(string finMessage) returns xml|error {
+    record {} customizedMessage = check swiftmt:parseSwiftMt(finMessage);
+    if customizedMessage is swiftmt:MT104Message {
+        return getMT104TransformFunction(customizedMessage);
+    }
+    xml swiftMessageXml = check xmldata:toXml(customizedMessage);
+    string messageType = (swiftMessageXml/**/<messageType>).data();
+    string validationFlag = (swiftMessageXml/**/<ValidationFlag>/<value>).data();
+    if validationFlag.length() > 0 {
+        isolated function (record {} message) returns record {}|error transformFunction = transformFunctionMap.get(messageType + validationFlag);
+        return xmldata:toXml(check transformFunction(customizedMessage));
+    }
+    isolated function (record {} message) returns record {}|error transformFunction = transformFunctionMap.get(messageType);
+    return xmldata:toXml(check transformFunction(customizedMessage), {textFieldName: "content"});
+}
+
+# Transforms the given SWIFT MT101 message to its corresponding ISO 20022 Pain.001 format.
+# 
+# This function extracts various fields from the SWIFT MT101 message and maps them to 
+# the appropriate ISO 20022 structure.
+#
+# + message - The parsed MT101 message as a record value.
+# + return - Returns the transformed ISO 20022 `Pain001Document` structure.
+# An error is returned if there is any failure in transforming the SWIFT message to ISO 20022 format.
+isolated function transformMT101(record {} message) returns record {}|error {
+    if message is swiftmt:MT101Message {
```

Comment:
```suggestion
    if message !is swiftmt:MT101Message {
        return error("Coversion of SWIFT MT 101 to ISO 20022 xml failed.");
    }
```
You can use early returns to avoid nested if statements where possible

Reply:
fixed in `601ea2a`